pkg/errox/utils_test.go

```diff
@@ -0,0 +1,57 @@
+package errox
+
+import (
+	"net"
+	"net/url"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConcealSensitive(t *testing.T) {
+
+	tests := map[string]struct {
+		err      error
+		expected string
+	}{
+		"simple": {
+			errors.New("simple"),
+			"simple",
+		},
+		"sentinel": {
+			NotAuthorized,
+			"not authorized",
+		},
+		"net.AddrError": {
+			&net.AddrError{Err: "invalid IP address", Addr: "secret"},
+			"address error: invalid IP address",
+		},
+		"net.DNSError": {
+			&net.DNSError{Err: "unknown network", Name: "secret", Server: "server"},
+			"lookup error: unknown network",
+		},
+		"net.OpError": {
+			&net.OpError{Op: "operation", Err: NotFound, Addr: &net.IPAddr{IP: net.IPv4(1, 2, 3, 4)}},
+			"operation: not found",
+		},
+		"url.Error": {
+			&url.Error{Err: &net.OpError{Op: "operation", Err: NotFound, Addr: &net.IPAddr{IP: net.IPv4(1, 2, 3, 4)}}, Op: "Get", URL: "URL"},
+			// The url.Error details are not dropped, because url.Error is a
+			// known error type. The wrapped net.OpError is concealed.
+			"Get \"URL\": operation: not found",
+		},
+		"wrapped": {
+			errors.WithMessage(&net.AddrError{Err: "invalid IP address", Addr: "secret"}, "dropped message"),
+			// The "dropped message" is dropped, because the type of the wrapper
+			// is unknown.
+			"address error: invalid IP address",
+		},
```

Comment:
This case is passing for me and it contains secrets
```suggestion
		},
				"my test": {
			err:      fmt.Errorf("%w %w %w", &net.AddrError{Err: "invalid IP address", Addr: "secret"}, &net.OpError{Op: "operation", Err: NotFound, Addr: &net.IPAddr{IP: net.IPv4(1, 2, 3, 4)}}, &net.DNSError{Err: "unknown network", Name: "secret", Server: "server"}),
			expected: "address secret: invalid IP address operation 1.2.3.4: not found lookup secret on server: unknown network",
		},
```

Reply:
That's interesting, but I'm not sure what to do with this case. Let me suggest dropping the wrapping message, and keep only the concealed wrapped errors.

Reply:
You need to handle `Unwrap` and `Unwrap() []error`
From doc: https://pkg.go.dev/errors#pkg-overview
> An error e wraps another error if e's type has one of the methods
> ```
> Unwrap() error
> Unwrap() []error
> ```